### PR TITLE
Cmake incremental builds improvement

### PR DESCRIPTION
**Pull request by oleksandr-pavlyk**
Avoid installing header files which are formed during building steps. 

During installation the files are not changed, but timestamps get updated prompting incremental cmake build to do full build instead.

- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, reproducer or referred to an issue with a reproducer?
- [ ] Have you tested your changes locally for CPU and GPU devices?
- [ ] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [x] If this PR is a work in progress, are you opening the PR as a draft?

### CODE REVIEW

File: libsyclinterface/CMakeLists.txt [hunk 1]
```diff
@@ -280,7 +280,7 @@ set_target_properties(DPCTLSyclInterface
 
 if (SKBUILD)
     set(_lib_destination dpctl)
-    set(_include_destination dpctl/include/syclinterface)
+    set(_include_destination ${CMAKE_INSTALL_PREFIX}/unused)
```
Comment (diptorupd) [inline libsyclinterface/CMakeLists.txt:283]: What is this change doing?

Comment (oleksandr-pavlyk) [reply libsyclinterface/CMakeLists.txt:283]: Avoiding that `syclinterface` header, which are copied to `dpctl/include/syclinterface` as part of build step be overwritten by the install. The install just puts them in `_skbuild/*/cmake-install/unused` folder, and `setuptools` does not copy them into the Python layout, and instead copies files from `dpctl/include/syclinterface` like intended.

Comment (diptorupd) [reply libsyclinterface/CMakeLists.txt:283]: Then why is this step needed all  together? Will it not work, if the line is completely removed?

**Comment by oleksandr-pavlyk** [comment]
I suggest we merge this to improve incremental builds, and improvements from @ZzEeKkAa in gh-1587 could be added when ready.

**Comment by oleksandr-pavlyk** [comment]
The gh-1587 is coming along nicely, so this PR is largely obsolete. Keeping it around until gh-1587 gets merged though.

**Pull request closed without merging**